### DISCUSSION TITLE
[feature request] Option to decrypt alert-notification integrationKey fields and expose via API

### DISCUSSION THREAD

**Discussion by adsail** [grafana/grafana#43813] [Alerting]
**What would you like to be added**:

As of Grafana 7.2, Alert Notification Channel `integrationKey`s (i.e. PagerDuty API Keys) are `secureFields` and thus not accessible via the Grafana API. 

Create a feature flag to offer the option to decrypt & expose `integrationKey` and other relevant `securefields` via the Grafana API with an admin API key.

**Why is this needed**:

This is fragmenting our process for configuring/managing notification channels between Grafana versions and rendering an external alerting engine to become unusable since it can no longer retrieve `integrationKey`s via the API.

**Comment by zuchka** [comment]
> This is fragmenting our process for configuring/managing notification channels between Grafana versions

Hi @adsail, thanks for submitting this feature request. One follow up question. What other versions are you using? 8.x.x? is the NGALERT feature toggle enabled? In any case, thanks for this!

**Comment by adsail** [comment]
@zuchka Thanks for the speedy response! We're attempting an upgrade from 6.5.2 to 7.5.3. Looks like `ngalert` only pertains to 8.x.x.

**Comment by zuchka** [comment]
Copy that--thanks for clarifying. I only asked because the next-gen alerting is a radical improvement on Grafana's earlier alerting system, and I have no idea what impact that might have on these APIs (it's all still in development...)  more soon!

**Comment by adsail** [comment]
Understood. If this feature request were to be taken up, we would certainly look towards 8.x.x as that is presumably where the feature would be implemented.

And that would enable us to explore next-gen alerting as well!

**Comment by yuri-tceretian** [comment]
I think this would be a useful feature. We can add a query parameter to API requests, for example, `show_secrets`, and support it only for Admin users. UI can use a regular request and display an `eye` icon beside the field that contains a secret. Once, the user clicks on it, UI sends the same request but with the query parameter and displays the result.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
